chainstate/src/detail/chainstateref.rs

```diff
@@ -528,25 +530,35 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
         let mut tx_verifier =
             TransactionVerifier::new(&self.db_tx, utxo_view.derive_cache(), self.chain_config);
 
-        tx_verifier.connect_transaction(
+        let reward_fees = tx_verifier.connect_transactable(
             BlockTransactableRef::BlockReward(block),
             spend_height,
             &median_time_past,
             blockreward_maturity,
         )?;
+        debug_assert!(reward_fees.is_none());
 
         // TODO: add a test that checks the order in which txs are connected
-        for (tx_num, _tx) in block.transactions().iter().enumerate() {
-            tx_verifier.connect_transaction(
-                BlockTransactableRef::Transaction(block, tx_num),
-                spend_height,
-                &median_time_past,
-                blockreward_maturity,
-            )?;
-        }
+        let total_fees = block
+            .transactions()
+            .iter()
+            .enumerate()
```

Comment:
Nitpick: I find the usage of `enumerate` here slightly misleading. You can use `0..block.transactions().len().map(...)` instead.

Reply:
@TheQuantumPhysicist I'm not insisting, but for me `enumerate` is a clear sign that we need to have an index in addition to the item we are iterating over, so I have spent some time checking why it is ignored. By the way, I don't see many possibilities for an error in the `0..len` range.

Reply:
Sure, when you know 0..len works, you do it. But then you get someone else modifying the code for some reason, rewriting it, and doing 0..len-1 reasoning that the length of the vector is len-1, and that happened here in Mintlayer a couple of months ago. My solution: Manual ranges only when we must.

I get your view on enumerate and you're right, but I'm more than happy to abuse it in the interest of avoiding a bug. This whole software is pivoted on security and the best practices we can come up with to reduce the probability of random bugs.